### PR TITLE
fixes heatmap API edge case

### PR DESCRIPTION
**Pull request by rithviknishad**
## Proposed Changes

- Fixes heatmap API 

In an edge case, `booked_slots` also includes "yesterdays" slot even though FE/client never asked for it. Hence throws KeyError since that day doesn't exist in `response_days`

## Merge Checklist

- [ ] Tests added/fixed
- [ ] Update docs in `/docs`
- [ ] Linting Complete
- [ ] Any other necessary step

_*Only PR's with test cases included and passing lint and test pipelines will be reviewed*_

@ohcnetwork/care-backend-maintainers @ohcnetwork/care-backend-admins

## Summary by CodeRabbit

- **Bug Fixes**
	- Improved accuracy of slot availability calculations.
	- Enhanced date-based filtering for scheduling statistics.

- **New Features**
	- Added advanced slot calculation method to improve scheduling precision.

### CODE REVIEW

File: care/emr/api/viewsets/scheduling/availability.py [hunk 1]
```diff
@@ -269,8 +269,8 @@ def availability_stats(self, request, *args, **kwargs):
 
         booked_slots = (
             TokenSlot.objects.filter(
-                start_datetime__lte=request_data.to_date,
-                end_datetime__gte=request_data.from_date,
+                start_datetime__date__lte=request_data.to_date,
```
Comment (rithviknishad) [inline care/emr/api/viewsets/scheduling/availability.py]:
```suggestion
                start_datetime__lte=request_data.to_date,
```

**Review by vigneshhari — APPROVED** [review]
